R/preprocess-pds-trips.R

```diff
@@ -0,0 +1,62 @@
+#' Pre-process Pelagic Data System trips
+#'
+#' Downloads raw structured data from cloud storage services and pre-process
+#' into a binary format that is easier to deal with in R.
+#'
+#' This function downloads the landings data from a given version (specified in
+#' the config file `conf.yml`. The parameters needed are:
+#'
+#' ```
+#' pds:
+#'   trips:
+#'     token:
+#'     secret:
+#'     file_prefix:
+#'   version:
+#'     preprocess:
+#' storage:
+#'   storage_name:
+#'     key:
+#'     options:
+#'       project:
+#'       bucket:
+#'       service_account_key:
+#' ```
+#'
+#' Progress through the function is tracked using the package *logger*.
+#'
+#' @param log_threshold
+#' @inheritParams ingest_pds_trips
+#' @keywords workflow
+#' @return no outputs. This function is used for it's side effects
+#' @export
+#'
+preprocess_pds_trips <- function(log_threshold = logger::DEBUG){
+
+  pars <- read_config()
+
+  pds_trips_csv <- cloud_object_name(prefix = pars$pds$trips$file_prefix,
+                                     provider = pars$storage$google$key,
+                                     extension = "csv",
+                                     version = pars$pds$trips$version$preprocess,
+                                     options = pars$storage$google$options)
+
+  logger::log_info("Retrieving {pds_trips_csv}")
+  download_cloud_file(name = pds_trips_csv,
+                      provider = pars$storage$google$key,
+                      options = pars$storage$google$options)
+  pds_trips_raw <- readr::read_csv(
+    file = pds_trips_csv,
+    col_types = readr::cols(.default = readr::col_character()))
```

Comment:
I'm thinking that maybe it's a good idea to also go over the field types in the data frame here? What do you think? If we convert things like time-dates (Start, End, etc) and integer/doubles (range, duration, etc) we might not need to do it several times later in the pipeline.

Reply:
You are right Fer, do you know what time zone have time columns?

Reply:
Hmm, not sure... I'll check with the PDS guys too but maybe we can have a guess by looking at the distribution of times? If we see a lot of activity after midnight it means we might have the wrong time zone :P